tests/libcxx/lit.site.cfg.in

```diff
@@ -1,25 +1,40 @@
 # Copyright (c) Microsoft Corporation.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
+import os
 import site
 site.addsitedir("@STL_TEST_UTILS_DIR@")
+site.addsitedir(os.path.join("@LIBCXX_SOURCE_DIR@", "utils"))
+import stl.test.config
+import stl.test.features
+import stl.test.file_parsing
+import stl.test.format
+import stl.test.params
 
-config.configuration_variant       = "stl"
-config.cxx_archive_root            = "@CMAKE_ARCHIVE_OUTPUT_DIRECTORY@"
-config.cxx_headers                 = "@STL_TESTED_HEADERS_DIR@"
-config.cxx_library_root            = "@CMAKE_LIBRARY_OUTPUT_DIRECTORY@"
-config.cxx_runtime_root            = "@CMAKE_RUNTIME_OUTPUT_DIRECTORY@"
-config.envlst_path                 = "@LIBCXX_ENVLST@"
-config.expected_results_list_path  = "@LIBCXX_EXPECTED_RESULTS@"
-config.format_name                 = "LibcxxTestFormat"
-config.include_dirs                = ["@LIBCXX_SOURCE_DIR@/test/support"]
-config.libcxx_obj_root             = "@LIBCXX_TEST_OUTPUT_DIR@"
-config.msvc_toolset_libs_root      = "@TOOLSET_LIB@"
-config.stl_build_root              = "@STL_BUILD_ROOT@"
-config.stl_src_root                = "@STL_SOURCE_DIR@"
-config.target_arch                 = "@VCLIBS_TARGET_ARCHITECTURE@"
-config.test_exec_root              = "@LIBCXX_TEST_OUTPUT_DIR@"
-config.test_source_root            = "@LIBCXX_SOURCE_DIR@/test"
-config.test_subdirs                = ["@LIBCXX_SOURCE_DIR@/test/std"]
+config.envlst_path      = '@LIBCXX_ENVLST@'
+config.name             = 'libc++'
+config.suffixes         = ['[.]pass[.]cpp$', '[.]fail[.]cpp$']
+config.test_exec_root   = '@LIBCXX_TEST_OUTPUT_DIR@'
+config.test_format      = stl.test.format.LibcxxTestFormat()
+config.test_source_root = '@LIBCXX_SOURCE_DIR@/test/std'
 
-lit_config.load_config(config, "@CMAKE_CURRENT_SOURCE_DIR@/lit.cfg")
+lit_config.expected_results = getattr(lit_config, 'expected_results', dict())
+lit_config.include_dirs     = getattr(lit_config, 'include_dirs', dict())
+lit_config.library_dirs     = getattr(lit_config, 'library_dirs', dict())
+lit_config.test_subdirs     = getattr(lit_config, 'test_subdirs', dict())
+
+lit_config.expected_results[config.name] = stl.test.file_parsing.parse_result_file('@LIBCXX_EXPECTED_RESULTS@')

```

Comment:
Something is wrong with libcxx's expected_results (not specifically this line, just had to comment somewhere). I'm seeing:

```
S:\GitHub\STL\out\build\x86>python tests\utils\stl-lit\stl-lit.py ..\..\..\llvm-project\libcxx\test ..\..\..\tests\std ..\..\..\tests\tr1 -o testing_x86.log
[...]
Testing Time: 6625.84s
  Unsupported      :  1895
  Passed           : 22816
  Expectedly Failed:    32
  Failed           :  1203
```
It looks like most tests marked `FAIL` aren't being treated as expected failures, and tests marked `SKIPPED` aren't being skipped (in particular, `rethrow_if_nested.pass.cpp` is one of the tests that displays a crash dialog box on my machine).